lib/blacklight/path.rb

```diff
@@ -0,0 +1,131 @@
+module Blacklight
+  # This class encapsulates the search state as represented by the query
+  # parameters namely: :f, :q, :page, :per_page and, :sort
+  class Path
```

Comment:
No, this is not a model in the sense that it doesn't represent a business object in a persistence store.